src/Apps/Settings/Routes/Purchases/Components/SettingsPurchasesRow.tsx

```diff
@@ -85,14 +96,61 @@ const getPaymentMethodText = (
   }
 }
 
+const getOrderActionButton = (
+  displayState: CommerceBuyerOrderOfferStateEnum,
+  orderId: string
+) => {
+  switch (displayState) {
+    case "PAYMENT_FAILED":
+      return (
+        <Button
+          // @ts-ignore
+          as={RouterLink}
+          to={`/orders/${orderId}/payment/new`}
+          variant="primaryBlack"
+          size="large"
+          width="50%"
+          minWidth={["240px", "200px"]}
+          my={[1, 0]}
+        >
+          Update Payment Method
+        </Button>
+      )
+    case "OFFER_RECEIVED":
```

Comment:
This function is called `getOrderActionButton()` but it is basically a component, right? What if we had a single map up above in [our consts](https://github.com/artsy/force/pull/14953/files#diff-a958048d8ace47200344fc15c5ba10bc9029a8250be78f2c2497a402111e3f75R41), something like this:
```tsx
const ORDER_STUFF: Record<CommerceBuyerOrderOfferStateEnum, {
  label: string,
  icon: JSX.Element
  ActionButton?: (props: { order: any }) => JSX.Element
} = { 
  APPROVED: { 
//...

// later
{ ActionButton && <ActionButton order={order} />
```

This way logic could live in one place for the most part.

Reply:
Good point. Will update. Thanks for suggestion. I think it did not make sense to begin with as we have only one special case for button. And I did not think of it as a change of pattern. But you are right - and it looks like we age going in a direction of having this button for more states. Totally legin change request.

Reply:
Ok. Circling back here and realizing that not sure if it's worth doing the bigger refactor you've proposed before we move this 'buyerDisplayState' or whatever we are going to call it to the server. In current state we need to massage this state and buyerAction data before display so went with the function suggestion here. Hopefully names are a bit more descriptive. 

Also ended up moving the function inline as realized the button to update payment was missing the tracking hook.

Also this brings a question - should we track a click to 'Respond to Counteroffer' button. Will check with analytics and do a follow up if so.